juju/juju
# Use the default of /var/lib/pebble for PEBBLE dir

Note that this keeps PEBBLE_SOCKET as /charm/container/pebble.socket, which is created by the K8s volume configuration -- no change there.

This updates go.mod to point Pebble to a branch on my fork which includes these two PRs:

* https://github.com/canonical/pebble/pull/12 - Create pebble dir on startup if it doesn't exist (if `--create-dirs` specified)
* https://github.com/canonical/pebble/pull/10 - Dynamic add-layer and get-plan APIs

I'll update this (or create another PR) to point to canonical/pebble once these are merged.

## QA steps

```
$ git clone https://github.com/benhoyt/snappass-test
$ cd snappass-test
$ charmcraft build
$ juju bootstrap microk8s
$ juju add-model snappass
$ juju deploy ../snappass-test/snappass-test.charm snappass --resource snappass-image=benhoyt/snappass-test --resource redis-image=redis
```

A bit later it should come up and `juju status` should say that unit `snappass/0` is active and idle, with message "snappass started". If you go to the IP address status shows at port 5000 in your browser (e.g., `http://10.1.147.102:5000/`) it should show the Snappass web interface.

You can also ensure that pebble and the workload is running correctly on the containers. For example:

```
$ microk8s.kubectl --namespace=snappass exec snappass-0 -it -c snappass -- bash
# cat /proc/1/cmdline | tr \\0 ' ' && echo
/charm/bin/pebble run --create-dirs --hold
# cat /proc/15/cmdline | tr \\0 ' ' && echo
/usr/local/bin/python /usr/local/bin/snappass
# ls -al /var/lib/pebble/default/
total 12
drwxr-xr-x 2 root root 4096 Mar  4 03:49 .
drwxr-xr-x 3 root root 4096 Mar  4 03:49 ..
-rw------- 1 root root  610 Mar  4 03:49 .pebble.state
```
